test/lit.cfg

```diff
@@ -2205,6 +2205,20 @@ config.environment['SWIFT_DEBUG_ENABLE_COW_CHECKS'] = 'false'
 # Add this to the command which runs an executable to enable COW checks in the swift runtime.
 config.substitutions.append(('%enable-cow-checking', TARGET_ENV_PREFIX + 'SWIFT_DEBUG_ENABLE_COW_CHECKS=true;'))
 
+# We add an expansion to ensure that migrator tests can search for the api diff
+# data dir from the host compiler toolchain instead of the resource dir of the
+# stdlib. Since we are using the host compiler, we would not have built any
+# tools meaning that we would not have generated the -api-diff-data-dir.
+#
+# In the case where we are not building against the host compiler, this just
+# expands to the empty string.
+api_diff_data_dir_flag = ''
+if config.testing_against_host_compiler:
+   migrator_dir = os.path.join(os.path.dirname(config.swiftc),
+                               '..', 'lib', 'swift', 'migrator')
```

Comment:
I was going to suggest:
```python
os.path.join(config.swift_lib_dir, 'migrator')
```

But I'm not actually sure if that's correct when testing against the host toolchain 🤔

Reply:
Yes. The swift lib dir is going to be where the newly built stdlib is I think. We want to be 100% sure it is the toolchain and swiftc is always going to be in the toolchain so I think it is reasonable.